client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -17,12 +27,58 @@ import {
 } from 'state/media/actions';
 
 import { registerHandlers } from 'state/data-layer/handler-registry';
+import {
+	dispatchFluxUpdateMediaItemSuccess,
+	dispatchFluxUpdateMediaItemError,
+} from 'state/media/actions/utils';
 
 /**
  * Module variables
  */
 const log = debug( 'calypso:middleware-media' );
 
+export function updateMedia( action ) {
+	const { siteId, item } = action;
+
+	return [
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/media/${ item.ID }`,
+				apiVersion: '1.1',
+				body: item,
+			},
+			action
+		),
+	];
+}
+
+export const updateMediaSuccess = ( { siteId }, mediaItem ) => ( dispatch ) => {
+	dispatch( receiveMedia( siteId, mediaItem ) );
+	dispatchFluxUpdateMediaItemSuccess( siteId, mediaItem );
+};
+
+export const updateMediaError = ( { siteId }, error ) => (/* dispatch, getState */) => {
+	dispatchFluxUpdateMediaItemError( siteId, error );
+};
+
+export const editMedia = ( action ) => {
+	const { siteId, item } = action;
+	const { ID: mediaId, ...rest } = item;
+
+	return [
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/media/${ mediaId }/edit`,
+				apiVersion: '1.1',
+				formData: toPairs( rest ),
```

Comment:
I'm not sure about this one. Apparently it seems to work correctly but `wpcom` uses a rather complex algorithm, [see here](https://github.com/Automattic/wp-calypso/blob/master/packages/wpcom.js/src/lib/site.media.js#L15-L54).